providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -420,6 +420,12 @@ static int dh_validate(const void *keydata, int selection, int checktype)
     if ((selection & DH_POSSIBLE_SELECTIONS) == 0)
         return 1; /* nothing to validate */
 
+#if defined(FIPS_MODULE) && defined(OPENSSL_NO_FIPS186_4_FFC)
+    /*
+     * Block operations on keys with FIPS 186-4 params.
+     */
+    selection |= OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS;
```

Comment:
I dont think you should be changing the selection here..

Reply:
Ok but will call DH_check_ex(dh) instead, as I don't see public/private checks later on call into that. And for acvp one needs to reject invalid keys.